common/persistence/nosql/nosqlExecutionStore.go

```diff
@@ -0,0 +1,831 @@
+	var nosqlTimerTasks []*nosqlplugin.TimerTask
+	var err error
+
+	// 1. current
+	if currentWorkflow != nil {
+		mutateExecution, err = d.prepareUpdateWorkflowExecutionRequestWithMapsAndEventBuffer(currentWorkflow)
+		if err != nil {
+			return err
+		}
+		nosqlTransferTasks, nosqlCrossClusterTasks, nosqlReplicationTasks, nosqlTimerTasks, err = d.prepareNoSQLTasksForWorkflowTxn(
+			domainID, workflowID, currentWorkflow.ExecutionInfo.RunID,
+			currentWorkflow.TransferTasks, currentWorkflow.CrossClusterTasks, currentWorkflow.ReplicationTasks, currentWorkflow.TimerTasks,
+			nosqlTransferTasks, nosqlCrossClusterTasks, nosqlReplicationTasks, nosqlTimerTasks,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	// 2. reset
+	resetExecution, err = d.prepareResetWorkflowExecutionRequestWithMapsAndEventBuffer(&resetWorkflow)
+	if err != nil {
+		return err
+	}
+	nosqlTransferTasks, nosqlCrossClusterTasks, nosqlReplicationTasks, nosqlTimerTasks, err = d.prepareNoSQLTasksForWorkflowTxn(
+		domainID, workflowID, resetWorkflow.ExecutionInfo.RunID,
+		resetWorkflow.TransferTasks, resetWorkflow.CrossClusterTasks, resetWorkflow.ReplicationTasks, resetWorkflow.TimerTasks,
+		nosqlTransferTasks, nosqlCrossClusterTasks, nosqlReplicationTasks, nosqlTimerTasks,
+	)
+	if err != nil {
+		return err
+	}
+
+	// 3. new
+	if newWorkflow != nil {
+		insertExecution, err = d.prepareCreateWorkflowExecutionRequestWithMaps(newWorkflow)
+
+		nosqlTransferTasks, nosqlCrossClusterTasks, nosqlReplicationTasks, nosqlTimerTasks, err = d.prepareNoSQLTasksForWorkflowTxn(
+			domainID, workflowID, newWorkflow.ExecutionInfo.RunID,
+			newWorkflow.TransferTasks, newWorkflow.CrossClusterTasks, newWorkflow.ReplicationTasks, newWorkflow.TimerTasks,
+			nosqlTransferTasks, nosqlCrossClusterTasks, nosqlReplicationTasks, nosqlTimerTasks,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	shardCondition := &nosqlplugin.ShardCondition{
+		ShardID: d.shardID,
+		RangeID: request.RangeID,
+	}
+
+	err = d.db.UpdateWorkflowExecutionWithTasks(
+		ctx, currentWorkflowWriteReq,
+		mutateExecution, insertExecution, resetExecution,
+		nosqlTransferTasks, nosqlCrossClusterTasks, nosqlReplicationTasks, nosqlTimerTasks,
+		shardCondition)
+	return d.processUpdateWorkflowResult(err, request.RangeID)
+}
+
+func (d *nosqlExecutionStore) DeleteWorkflowExecution(
+	ctx context.Context,
+	request *p.DeleteWorkflowExecutionRequest,
+) error {
+	err := d.db.DeleteWorkflowExecution(ctx, d.shardID, request.DomainID, request.WorkflowID, request.RunID)
+	if err != nil {
+		return convertCommonErrors(d.db, "DeleteWorkflowExecution", err)
+	}
+
+	return nil
+}
+
+func (d *nosqlExecutionStore) DeleteCurrentWorkflowExecution(
+	ctx context.Context,
+	request *p.DeleteCurrentWorkflowExecutionRequest,
+) error {
+	err := d.db.DeleteCurrentWorkflow(ctx, d.shardID, request.DomainID, request.WorkflowID, request.RunID)
+	if err != nil {
+		return convertCommonErrors(d.db, "DeleteWorkflowCurrentRow", err)
```

Comment:
nit: do we want to just use the same as the method name? DeleteCurrentWorkflow

Reply:
yeah we should do that.